### ISSUE TITLE
dropdownbutton in modal: esc closes modal when menu is open

### ISSUE THREAD

**Issue by acthp** [react-bootstrap/react-bootstrap#486]
Normally the escape key closes a drop-down. However, inside a modal it will close the modal, which is undesirable in my case (and probably most others).  I expect this is because it's not doing a ev.stopPropagation(), so the modal (or anything else above it) is also reacting to the escape key.

Perhaps it should be configurable, if there's some use case where you'd want a containing element to also react to the escape key when the menu closes, but this seems unlikely. Probably in that case you'd instead want a 'close' event.

**Comment by acthp** [comment]
btw, if an element in the dropdown has focus, it's possible to work around this by binding an onKeyUp handler which calls ```ev.nativeEvent.stopImmediatePropagation()```, and ```this.refs.dropdown.setDropDownState(false)```. Not sure why ```ev.stopPropagation()``` doesn't work here.

**Comment by mtscout6** [comment]
Definitely a bug, PR welcome to address the issue.

**Comment by AlexKVal** [comment]
I believe that it is also partly an issue with Modal, isn't it ?

**Comment by AlexKVal** [comment]
Ah.. Sorry.
It is already somehow addressed https://github.com/react-bootstrap/react-bootstrap/issues/526
```
Escape
 [x] With focus on the drop-down menu, pressing Escape closes the menu and returns focus to the button. Resolves #486
```
